src/library_fetchfs.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * @license
+ * Copyright 2013 The Emscripten Authors
+ * SPDX-License-Identifier: MIT
+ */
+
+mergeInto(LibraryManager.library, {
+  $FETCHFS__deps: ['$stringToUTF8OnStack', 'wasmfs_create_fetch_backend'],
+  $FETCHFS: {
+    createBackend(opts) {
+      return _wasmfs_create_fetch_backend(stringToUTF8OnStack(opts.base_url));
+    }
+  },
+});
+
+if (!WASMFS) {
```

Comment:
Should these checks be `#if` preprocessor checks?

Reply:
I did consider using `#if`, but was a little unsure how to trigger the error if the `library_fetchfs.js` file actually got linked in. Does calling `error()` inside a top-level `#if` (As below) have the potential to cause any problems? 
```
#if !WASMFS
error("using -lfetchfs is not currently supported in WasmFS."
#endif
```

Reply:
This are ok as normal JS. We use the preprocessor to filter what code is emitted in the output JS, but this code is not in the output JS - unlike say `preamble.js` which is in the output, this is a JS library file which is executed. So it's ok to use normal JS execution to check things like `if (..) {..`